src/sentry/snuba/metrics/naming_layer/mapping.py

```diff
@@ -63,9 +68,12 @@ def get_public_name_from_mri(internal_name: Union[TransactionMRI, SessionMRI, st
         internal_name = internal_name.value
     assert isinstance(internal_name, str)
 
-    try:
+    if internal_name in MRI_TO_NAME:
         return MRI_TO_NAME[internal_name]
-    except KeyError:
+    elif (match := MRI_SCHEMA_REGEX.match(internal_name)) is not None:
```

Comment:
You need to check if it is an instance of `TransactionMRI` and that it is a distribution

Reply:
For the second check you basically need to check that the dataset belongs to `CUSTOM_MEASUREMENTS_DATASETS` https://github.com/getsentry/sentry/blob/dec6a2d3a89b304530627070b660c5fee61e2444/src/sentry/snuba/metrics/datasource.py#L204